src/structures/InteractionCollector.js

```diff
@@ -33,21 +33,28 @@ class InteractionCollector extends Collector {
 
     /**
      * The message from which to collect interactions, if provided
-     * @type {?Message}
+     * @type {?Snowflake}
      */
-    this.message = options.message ?? null;
+    this.messageId = options.message?.id ?? options.message ?? null;
```

Comment:
might it be worth having some sort of `resolveId` function for resolving the message ID?